commons/ava_testsuite/byzantine_gecko_network_tests.go

```diff
@@ -0,0 +1,115 @@
+	castedNetwork := network.(ava_networks.TestGeckoNetwork)
+	for i := 0; i < 4; i++ {
+		byzHighLevelClient, err := addServiceIdAsValidator(castedNetwork, i, BYZANTINE_USERNAME, BYZANTINE_PASSWORD, SEED_AMOUNT, STAKE_AMOUNT)
+		if err != nil {
+			context.Fatal(stacktrace.Propagate(err, "Failed to add Byzantine service as a validator."))
+		}
+		byzClient := byzHighLevelClient.GetLowLevelClient()
+		currentStakers, err := byzClient.PChainApi().GetCurrentValidators(nil)
+		if err != nil {
+			context.Fatal(stacktrace.Propagate(err, "Could not get current stakers."))
+		}
+		logrus.Debugf("Number of current stakers: %d", len(currentStakers))
+	}
+	availabilityChecker, err := castedNetwork.AddService(NORMAL_NODE_CONFIG_ID, 4)
+	if err != nil {
+		context.Fatal(stacktrace.Propagate(err, "Failed to add normal node with high quorum and sample to network."))
+	}
+	err = availabilityChecker.WaitForStartup()
+	if err != nil {
+		context.Fatal(stacktrace.Propagate(err, "Failed to wait for startup of normal node."))
+	}
+	stakerHighLevelClient, err := addServiceIdAsValidator(castedNetwork, 4, STAKER_USERNAME, STAKER_PASSWORD, SEED_AMOUNT, STAKE_AMOUNT)
+	if err != nil {
+		context.Fatal(stacktrace.Propagate(err, "Failed to add Normal service as a validator."))
+	}
+	stakerClient := stakerHighLevelClient.GetLowLevelClient()
+	currentStakers, err := stakerClient.PChainApi().GetCurrentValidators(nil)
+	if err != nil {
+		context.Fatal(stacktrace.Propagate(err, "Could not get current stakers."))
+	}
+	logrus.Debugf("Number of current stakers: %d", len(currentStakers))
+	actualNumStakers := len(currentStakers)
+	expectedNumStakers := 10
+	context.AssertTrue(actualNumStakers == expectedNumStakers, stacktrace.NewError("Actual number of stakers, %v, != expected number of stakers, %v", actualNumStakers, expectedNumStakers))
+}
+func (test StakingNetworkUnrequestedChitSpammerTest) GetNetworkLoader() (testsuite.TestNetworkLoader, error) {
+	return getByzantineNetworkLoader(map[int]int{
+		0:           BYZANTINE_CONFIG_ID,
+		1:           BYZANTINE_CONFIG_ID,
+		2:           BYZANTINE_CONFIG_ID,
+		3:           BYZANTINE_CONFIG_ID,
+	}, test.unrequestedChitSpammerImageName, test.normalImageName)
+}
+func (test StakingNetworkUnrequestedChitSpammerTest) GetTimeout() time.Duration {
+	return 720 * time.Second
+}
+
+
+
+// =============== Helper functions =============================
+
+/*
+Args:
+	desiredServices: Mapping of service_id -> configuration_id for all services *in addition to the boot nodes* that the user wants
+*/
+func getByzantineNetworkLoader(desiredServices map[int]int, byzantineImageName *string, normalImageName *string) (testsuite.TestNetworkLoader, error) {
+	serviceConfigs := map[int]ava_networks.TestGeckoNetworkServiceConfig{
+		NORMAL_NODE_CONFIG_ID: *ava_networks.NewTestGeckoNetworkServiceConfig(true, ava_services.LOG_LEVEL_DEBUG, normalImageName, 6, 8),
```

Comment:
this is actually relying on `NORMAL_NODE_CONFIG_ID` constant that lives in the other test file!! let's not do that please - this file should contain all the consts that he uses